lake/commit/transaction.go

```diff
@@ -17,26 +17,26 @@ type Transaction struct {
 	Actions []actions.Interface `zng:"actions"`
 }
 
-func newTransaction(id ksuid.KSUID, capacity int) Transaction {
-	return Transaction{
+func newTransaction(id ksuid.KSUID, capacity int) *Transaction {
+	return &Transaction{
 		ID:      id,
 		Actions: make([]actions.Interface, 0, capacity),
 	}
 }
 
-func NewCommitTxn(id ksuid.KSUID, date nano.Ts, author, message string, segments []segment.Reference) Transaction {
+func NewCommitTxn(id ksuid.KSUID, date nano.Ts, author, message string, segments []segment.Reference) *Transaction {
```

Comment:
Most (all?) of the new code here uses `[]*segment.Reference`, and we should eventually use that instead of `[]segment.Reference` everywhere.

Reply:
Yes, I realized this and we should keep this mind as we continue to develop these pieces.

Reply:
This was partially a motivation for Segments.

Reply:
We can clean this up in a subsequent PR.